packages/forms/src/UIForm/fields/File/File.component.js

```diff
@@ -0,0 +1,157 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { translate } from 'react-i18next';
+import FieldTemplate from '../FieldTemplate';
+import { generateDescriptionId, generateErrorId } from '../../Message/generateId';
+
+import { I18N_DOMAIN_FORMS } from '../../../constants';
+import theme from './File.scss';
+
+/**
+ * Extract the file name from the data url
+ * @param {String} value The base64 value of the file.
+ * Looks like `data:text/xml;name=test.xml;base64,PD94bWwgdmVyc2l...`
+ * @return {String} The file name, for exemple: `test.xml`.
+ */
+function getFileName(value) {
+	if (value && value.indexOf(';name=') !== -1) {
+		return value.slice(value.indexOf(';name=') + 6, value.indexOf(';base64,'));
+	}
+	return '';
+}
+
+/**
+ * Add the file name to the data url.
+ * @param {String} value The base64 value of the file.
+ * Looks like `data:text/xml;base64,PD94bWwgdmVyc2l...`
+ * @param {String} fileName The file name, for exemple `test.xml`.
+ * @return {string} the base 64 encoding of the file with the file name within.
+ * Looks like `data:text/xml;name=test.xml;base64,PD94bWwgdmVyc2l...`
+ */
+function getBase64(value, fileName) {
+	if (value && value.indexOf(';name=') === -1) {
+		const fileNamePos = value.indexOf(';base64,');
+		if (fileNamePos !== -1) {
+			return [value.slice(0, fileNamePos), ';name=', fileName, value.slice(fileNamePos)].join('');
+		}
+	}
+	return value;
+}
+
+class FileWidget extends React.Component {
+	constructor(props) {
+		super(props);
+		this.onChange = this.onChange.bind(this);
+		// Extract file name from form properties
+		this.state = { fileName: getFileName(props.value) };
+	}
+
+	onChange = event => {
```

Comment:
Could be a regular function. 
```suggestion
	onChange(event) {
```